pkg/plugins/virtual/virtual_plugin.go

```diff
@@ -84,7 +84,18 @@ func (p *VirtualPlugin) Apply() error {
 	glog.Infof("virtual-plugin Apply(): desiredState=%v", p.DesireState.Spec)
 
 	if p.LoadVfioDriver == loading {
-		if err := utils.LoadKernelModule("vfio_pci"); err != nil {
+		// In virtual deployments of Kubernetes where the underlying virtualization platform does not support a virtualized iommu
+		// the VFIO PCI driver needs to be loaded with a special flag.
+		// This is the case for OpenStack deployments where the underlying virtualization platform is KVM.
+		// NOTE: if VFIO was already loaded for some reason, we will not try to load it again with the new options.
+		kernel_args := []string{"enable_unsafe_noiommu_mode=1"}
```

Comment:
>the virtual plugin is only used for OpenStack (now) and enable_unsafe_noiommu_mode is always needed on OpenStack because we don't support virtualized IOMMU. So I think it's safe to enable it by default for now.

i have limited expeirence with SNO on virtualized env with openstack where this plugin is used. 
we may need later on to expose the option to configure vfio kernel args policy.

for now if its always needed then i think its ok.

Reply:
yes it's always needed and I don't foresee it changing since it'll require *huge* changes in OpenStack to support this.